### ISSUE TITLE
Page reload not working (organization_hint cookie missing) with auth0-spa-js > 2.0.7

### ISSUE THREAD

**Issue by christianblust** [auth0/auth0-angular#465]
### Checklist

- [X] The issue can be reproduced in the [auth0-angular sample app](https://github.com/auth0-samples/auth0-angular-samples/tree/master/Sample-01) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/auth0-angular#readme), [Examples](https://github.com/auth0/auth0-angular/blob/master/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-angular/blob/master/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/auth0-angular/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-angular/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Hello everyone,

I'm in the process of integrating Auth0 into a backend API paired with an Angular frontend. I must admit, I'm not particularly adept with Angular or frontend development, but I've been closely following the provided quickstart guide.

I've encountered an issue related to page reloads in my custom implementation. Specifically, when I use versions of auth0-spa-js beyond 2.0.8, the organization_hint cookie doesn't seem to be set. As a result, the organization parameter isn't appended to the /authorize call. Could this be causing my page reload issues?

Additionally, I've come across another challenge when setting useRefreshTokens: true. In the Chrome DevTools, I can see a refresh token being returned. Yet, when I refresh the page, the isAuthenticated state seems to be false. I also do not see any attemps for a refresh in the network tab. However, setting cacheLocation: 'localstorage' seems to resolve this, but I suspect this might not be the best configuration. Can anyone shed light on this?

Here's the configuration in my app.module for reference:

```
@NgModule({
    declarations: [
        AppComponent,
        HomeComponent,
        NavbarComponent
    ],
    imports: [
        BrowserModule,
        AppRoutingModule,
        AuthModule.forRoot({
            domain: "tenantdomain",
            clientId: "clientId",
            //useRefreshTokens: true,
            authorizationParams: {
                audience: "https://audience-api",
                redirect_uri: window.location.origin
            },
            httpInterceptor: {
                allowedList: [`api/*`],
            }
        })
    ],
    providers: [
        {
            provide: HTTP_INTERCEPTORS, useClass: AuthHttpInterceptor, multi: true
        },
        {provide: Window, useValue: window},
    ],
    bootstrap: [AppComponent]
})
export class AppModule {
}
```

Versions: `angular 16.2.0`, `auth0/auth0-angular 2.0.8` works, `2.1.0` and greater do not.

Any hints and general advice (regarding the refresh tokens) would be great!

### Reproduction

1. Download quickstart
2. Set spa version accordingly
3. Reload page

### auth0-angular version

2.2.1

### Angular version

16.2.0

### Which browsers have you tested in?

Chrome, Firefox

**Comment by frederikprijck** [comment]
I just tried the latest version of our SDK and the organization cookie is set correctly.

> auth0/auth0-angular 2.0.8 works, 2.1.0 and greater do not.

There is no auth0-angular 2.0.8, so I am abit confused what you mean here.

Additionally, this issue is about organizations, but I see no code how you set the organization in the SDK. Can you show how the organization is set to help understand what might be causing it? 

> Additionally, I've come across another challenge when setting useRefreshTokens: true. In the Chrome DevTools, I can see a refresh token being returned. Yet, when I refresh the page, the isAuthenticated state seems to be false. I also do not see any attemps for a refresh in the network tab. However, setting cacheLocation: 'localstorage' seems to resolve this, but I suspect this might not be the best configuration. Can anyone shed light on this?

Set `useRefreshTokensFallback` to true when not using local storage. But know this relies on third party cookies, so when these are being blocked we recommend to use custom domains (or local storage if custom domains isnt an option).

**Comment by christianblust** [comment]
Ah sorry for the typo there: I meant the spa version `auth0-spa-js 2.0.8`. So going greater than `2.0.8` is causing issues for me.

Currently I don't do anything to configure organizations in the sdk. It simply works. After a successful login into an organization the cookie `organization_hint` is created with the organizationId I logged into. When reloading the page this organization is appended to the authorize call. This does not happen (for me) for `auth0-spa-js` versions greater than `2.0.8`. I observe this behaviour without the use of refreshtokens. Should I configure anything for organizations? Didn't see this yet in examples.

Thanks for the hint regarding the localStorage in combination with the refresh-tokens, I will try that!

**Comment by frederikprijck** [comment]
> Should I configure anything for organizations? Didn't see this yet in examples.

I recommend checking out https://github.com/auth0/auth0-spa-js/blob/master/EXAMPLES.md#organizations and see if it helps, however, that shouldnt differ between 2.0.8 and 2.1.0.

I will do some additional investigation on my end.

**Comment by frederikprijck** [comment]
Thanks, I know what's causing it. You should be able to get it working by setting the organization id up front as per the examples.

However, I will fix this so that it works the same as 2.0.8 again.

**Comment by frederikprijck** [comment]
A new release is available for auth0-spa-js, can you update to v2.1.2 and see if it works for you now ?
